test/integration/tracked_watermark_buffer.h

```diff
@@ -116,9 +129,16 @@ class TrackedWatermarkBufferFactory : public Buffer::WatermarkFactory {
   using AccountToBoundBuffersMap =
       absl::flat_hash_map<BufferMemoryAccountSharedPtr,
                           absl::flat_hash_set<TrackedWatermarkBuffer*>>;
+  // Used to inspect all accounts tied to any buffer created from this factory.
   void inspectAccounts(std::function<void(AccountToBoundBuffersMap&)> func,
                        Server::Instance& server);
 
+  // Used to inspect the memory class to accounts within that class structure.
+  // This differs from inspectAccounts as that has all accounts bounded to an
+  // active buffer, while this might not track certain accounts (e.g. below
+  // thresholds.)
+  void inspectMemoryClasses(std::function<void(MemoryClassesToAccountsSet&)> func);
```

Comment:
Worth pointing out at as implemented, this is thread unsafe -- not an issue with any of the tests using it!